### DISCUSSION TITLE
different investment costs in different years

### DISCUSSION THREAD

**Discussion by zsalimian** [iiasa/message_ix#511] [MESSAGE modeling]
I have defined three techonologies with investment costs in MESSAGE-ix
base_inv_cost = {
    'node_loc': country,
    'year_vtg': model_horizon,
    'unit': 'USD/kW',
}

costs = {
    'cycled_ppl': 632,
    'wind_ppl':500,
    'grid':152.4,
}

but I would like to define investment costs for different years.
for instance, investment costs for 'cycled_ppl' is 632$ in 2010 and will be 600$ in 2040.
How to define different investment costs in different years?

**Comment by zsalimian** [comment]
If possible please explain about defining fixed cost too.

**Reply by behnam-zakeri** [reply]
Thanks @zsalimian for reaching out. You can define different _values_  for different years for a MESSAGEix parameter like `inv_cost`. Assuming that you are using the code based on MESSAGEix tutorials, what you need to do is just to add those values in a python _list_, with the same order of the years, when you are wrapping the input data. For example, if your `model_horizon` has four years, you can add different costs per each year for "cycled_ppl" as follows:
```
costs = {
'cycled_ppl': [632, 622, 612, 600],
'wind_ppl':500,
'grid':152.4,
}
```

**Comment by LauWien** [comment]
Dear @zsalimian, 
Since this is neither about (a) a bug or error in the MESSAGEix package nor (b) a requested feature in the software per se, I have converted this from an issue to a “Discussion”. As the documentation indicates, please only open issues if (a) or (b) is the case; otherwise use the Discussions feature.
